### ISSUE TITLE
Speed up unit tests

### ISSUE THREAD

**Issue by k-yle** [openstreetmap/iD#10408]
There are a lot of annoying aspects to our current unit test system:
 - Currently we run all tests in a headless chrome browser. This is painfully slow, and also redundant for the majority of tests cases which don't need a real browser environment
 - We use karma (see #8764), but it's deprecated and has performance problems
 - We run our tests using the compiled JS bundle (`dist/iD.js`), instead of importing the source code itself. This means it takes forever to test your changes. After every change, you have to run `npm run build:js`, restart `npm run test:spec` (which restarts chrome), then reattach the debugger, and then wait for ages for the results. 
 - There is no autocomplete for the chai assertions (see #9805)

I think there are some simple changes that would make it much easier to write tests.

Based on a quick test, 1733 of our 2354 tests (73%) pass when running with [vitest](https://github.com/vitest-dev/vitest)+[jsdom](https://github.com/jsdom/jsdom), with barely any changes required. And it's about 10-20x faster 🎉

Therefore, I propose we introduce a lightweight test framework, and rename each test file based on which test framework it uses:

 - 73% of tests would keep the file extension `*.test.js`, an run with a lightweight framework, e.g. vitest or jest
 - 27% of test files could be renamed to `*.e2e.js`, and continue to use karma + a headless browser.

This would be a major improvement for the majority of tests which are not UI-related. There would be no improvement for the tests related to UI/rendering code.

**Comment by tyrasd** [comment]
Yeah, splitting off the tests which are not reliant on a full browser environment makes sense from a performance point of view. :+1: 

For the other mentioned issues, I've thought about migrating the test suite to esbuild + x instead of karma + precompiled js. This would also be required for upgrading some test libraries (e.g. fetch-mock #10346).

**Comment by tyrasd** [comment]
this is done, right @k-yle ?

**Comment by k-yle** [comment]
yeah, this should have been auto-closed when #10452 was merged
